packages/repository/test/unit/repositories/legacy-juggler-bridge.unit.ts

```diff
@@ -137,6 +142,18 @@ describe('DefaultCrudRepository', () => {
     expect(result.toJSON()).to.eql(note.toJSON());
   });
 
+  it('hides hidden properties', async () => {
+    const repo = new DefaultCrudRepository(Note, ds);
+    const note = await repo.create({
+      title: 't3',
+      content: 'c3',
+      secret: 'secret',
+    });
+    expect(note.secret).to.be.undefined();
+    const result = await repo.findById(note.id);
+    expect(result.secret).to.be.undefined();
```

Comment:
The `hidden/protected(secret)` properties are really interpreted as `writeonly` during serialization in LoopBack 3. They can be written but not read/search. Please note that a secret property can still be modified. For example:
```ts
repo.updateById(note.id, {secret: 'new-secret'});
```
But I agree with you that unconditionally removing them from CRUD is not ideal as we won't be able to support the use cases such as username/password verification. 

I propose the following:

1. Improve juggler so that options to control how to honor secret properties can be set at method level too in addition to datasource/model scope.

2. Update CRUD controller code template/base class to hide secret properties for REST requests.

3. For juggler.next, we need to reevaluate such property modifiers.